sys/cpp11-compat/cppsupport.cpp

```diff
@@ -144,4 +144,26 @@ void operator delete[](void* ptr, const std::nothrow_t&) noexcept {
     std::free(ptr);
 }
 
+extern "C" {
+#include <sys/time.h>
+#include "xtimer.h"
+
+/*
+ * This is a workaround for missing support in clang on OSX,
+ * the alias is not needed in native.
+ * Compare https://github.com/RIOT-OS/RIOT/issues/2336
+ */
+#ifndef BOARD_NATIVE
+void _gettimeofday(void)           __attribute__ ((weak, alias("cpp11_compat_gettimeofday")));
+#endif
+
+void cpp11_compat_gettimeofday(struct timeval *tp)
+{
+    timex_t now;
+    xtimer_now_timex(&now);
+    tp->tv_sec = now.seconds;
+    tp->tv_usec = now.microseconds;
+}
+}
```

Comment:
The bunch is now in the `cppsupport.cpp`, I think this is the proper place for it since its only used by cpp (by now) :)
@josephnoir @gebart any objections?